### PR TITLE
Task/02 interfaces

### PR DESCRIPTION
**Pull request by jaxin007**
Acceptance criteria for 1

- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  Interfaces for Product and Category based on [swagger](https://ma-backend-api.mocintra.com/docs#/categories/CategoriesController_getAll) are created and exported in separate file.
- [x]  Static header and footer are created according to [mobile](https://www.figma.com/file/mRAOMdJiAsD8fHSVnYwUl9/React-web-page?type=design&node-id=13-17490&mode=design&t=3Thf45kJJYMoxsc0-0) and [desktop](https://www.figma.com/file/MIKTcyr6SvspkZnIQ5nEPD/MA-Courses-2022-2023?type=design&node-id=2471-2253&mode=design&t=4DJ0BIdFZZ4mUgAn-0) designs.

Acceptance criteria for 1*

- [x]  Shared component SocialIcon is created and used for footer icons
- [x]  Link to social network and path to icon are passed to component by the typed props

### CODE REVIEW

File: src/components/TextWithRedirect/TextWithRedirect.component.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
```
Comment (tetiana-zhuchenko) [inline src/components/TextWithRedirect/TextWithRedirect.component.tsx]: You don't need to do this import. Could you delete it please?

File: src/components/IconWithRedirect/IconWithRedirect.component.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
```
Comment (tetiana-zhuchenko) [inline src/components/IconWithRedirect/IconWithRedirect.component.tsx]: You don't need to do this import. Could you delete it please?

File: src/components/Footer/Footer.component.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            <div className={styles.footerInfo}>
+                <div className={styles.line}></div>
+                <div className={styles.iconsBlock}>
+                    <IconWithRedirect icon={images.social.facebook} url={links.facebook}></IconWithRedirect>
+                    <IconWithRedirect icon={images.social.linkedin} url={links.linkedin}></IconWithRedirect>
+                    <IconWithRedirect icon={images.social.instagram} url={links.instagram}></IconWithRedirect>
+                </div>
+                <p>
+                    Made with 💗 on course{' '}
+                    <a href={links.google} className={styles.orangeLink}>
```
Comment (tetiana-zhuchenko) [inline src/components/Footer/Footer.component.tsx:19]: You may use the same link as in image below, but it's up to you

image

File: src/components/Header/header.module.css
Comment (tetiana-zhuchenko) [inline src/components/Header/header.module.css:1]: I think it's better to capitalize the module name "Header.module.css"

image

File: src/components/Footer/footer.module.css
Comment (tetiana-zhuchenko) [inline src/components/Footer/footer.module.css:1]: I think it's better to capitalize the module name "Footer.module.css"

image

File: src/models/category.model.ts
Comment (tetiana-zhuchenko) [inline src/models/category.model.ts:1]: It's a common practice to give the name "types" for your folder with types and interface. But it's up to you.

image

**Review by tetiana-zhuchenko — COMMENTED** [review]
Dima, you've made a very good job. Thanks for your time. Could you consider making some corrections?

File: src/models/category.model.ts
Comment (jaxin007) [reply src/models/category.model.ts:1]: I think for internal types the `types` name should be used. For interfaces, it's a common practice to name folder as a models or interfaces

File: src/constants/links.constant.ts
Comment (tetiana-zhuchenko) [inline src/constants/links.constant.ts:1]: The one more way to make links.constans
 
 export const links: <T> = [
    {
        name: 'Facebook',
        icon: facebookIcon,
        url: 'https://www.facebook.com/',
    },
    {
        name: 'Linkedin',
        icon: linkedinIcon,
        url: 'https://linkedin.com/',
    },
    {
        name: 'Instagram',
        icon: instagramIcon,
        url: 'https://www.instagram.com/',
    },
];

**Review by tetiana-zhuchenko — APPROVED** [review]

**Pull request merged**